src/windows/Animated.tsx

```diff
@@ -0,0 +1,101 @@
+/**
+* Animated.tsx
+*
+* Copyright (c) Microsoft Corporation. All rights reserved.
+* Licensed under the MIT license.
+*
+* RN Windows-specific implementation of Animated wrapper.
+*/
+
+import React = require('react');
+import RN = require('react-native');
+import RX = require('../common/Interfaces');
+import Types = require('../common/Types');
+import RXView from './View';
+import {Animated as AnimatedBase} from '../native-common/Animated';
+
+var ReactAnimatedView = RN.Animated.createAnimatedComponent(RXView, true);
+
+export class AnimatedView extends RX.AnimatedView {
+
+    private _animatedComponent: typeof ReactAnimatedView;
+
+    constructor(props: Types.AnimatedViewProps) {
+        super(props);
+    }
+
+    setNativeProps(props: Types.AnimatedViewProps) {
+        if (this._animatedComponent) {
+            if (!this._animatedComponent.setNativeProps) {
+                throw 'Component does not implement setNativeProps';
+            }
+            this._animatedComponent.setNativeProps(props);
+        }
+    }
+
+    render() {
+        return (
+            <ReactAnimatedView
+                ref={ this._onAnimatedComponentRef }
+                { ...this.props }
+            >
+                { this.props.children }
+            </ReactAnimatedView>
+        );
+    }
+
+    focus() {
+        if (this._animatedComponent && this._animatedComponent._component) {
+            this._animatedComponent._component.focus();
```

Comment:
>_component [](start = 36, length = 10)

What is `_component`? Why don't we need this indirection when implementing `setNativeProps`?

Reply:
_animatedComponent is a helper component used by the Animated package (you can check in react-native/Libraries/Animated/src/AnimatedImplementation.js, createAnimatedComponent).
It renders whatever you passed in (an RN.View in this case), with a ref function setting _component. So _component is the RN.View instance in this case.
Now, animatedComponent has a straight forwarder of setNativeProps to _component.setNativeProps, so no special indirection is needed there.
The "focus" method is outside this story.


---
In reply to: [159358021](https://github.com/Microsoft/reactxp/pull/404#discussion_r159358021) [](ancestors = 159358021)